pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,49 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+	if len(storageAccount) == 0 {
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
+			storageAccount = getStorageAccountName(fileShareAccountNamePrefix)
+			if location == "" {
+				location = az.controllerCommon.location
+			}
+			if storageType == "" {
+				storageType = defaultStorageAccountType
+			}
+
```

Comment:
If storageType is empty, and then it got set here.
Suppose Line#67 will always fail? For new created account or user-specified account, account.StorageType is empty according to Line#63.